File: Intervals/intervals0.java
import java.util.Scanner;

public class intervals0 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int k = scanner.nextInt();

        int[] hours = new int[24];

        for (int i = 0; i < n; i++) {
            int a = scanner.nextInt();
            int b = scanner.nextInt();

            for (int h = a; h < b; h++) {
                hours[h]++;
            }
        }

        int count = 0;

        for (int h = 0; h < 24; h++) {
            if (hours[h] >= k) {
                count++;
            }
        }

        System.out.println(count);
    }
}

File: Intervals/intervals0.c
#include <stdio.h>

int main() {
    int n, k;
    scanf("%d %d", &n, &k);

    int hours[24] = {0};

    for (int i = 0; i < n; i++) {
        int a, b;
        scanf("%d %d", &a, &b);

        for (int h = a; h < b; h++) {
            hours[h]++;
        }
    }

    int count = 0;

    for (int h = 0; h < 24; h++) {
        if (hours[h] >= k) {
            count++;
        }
    }

    printf("%d\n", count);
    return 0;
}
